src/main/java/ch/jalu/configme/beanmapper/propertydescription/BeanPropertyDescription.java

```diff
@@ -5,10 +5,11 @@
 import org.jetbrains.annotations.Nullable;
 
 /**
- * Represents a property within a bean class, as used in {@link ch.jalu.configme.beanmapper.MapperImpl}.
- * There, for instance, there is a {@link BeanDescriptionFactory} field responsible for creating bean descriptions.
+ * Represents a property within a bean class, as used by
+ * {@link ch.jalu.configme.beanmapper.instantiation.BeanInstantiation BeanInstantiation}. Bean property descriptions
+ * are provided by {@link BeanDescriptionFactory}.
  * <p>
- * Default implementation is {@link BeanPropertyDescriptionImpl}.
+ * Default implementation of this interface is {@link BeanFieldPropertyDescription}.
  */
 public interface BeanPropertyDescription {
```

Comment:
Still not fully happy with the names here. Is Description right? Should it be just `BeanComponent`? 
What about the `BeanDescriptionFactory`?